### PR TITLE
Plan view helper

### PR DESCRIPTION
**Pull request by stefanotorresi**
Forward port of SUSE-Enceladus/blue-horizon#163.

Still in draft because it depends on SUSE-Enceladus/blue-horizon#167 and it's going to need a rebase as soon as that is merged.

### CODE REVIEW

**Comment by stefanotorresi** [comment]
here is an updated screenshot:

Screenshot from 2020-11-06 17-53-05

I will still defer merging because we might need to move this away from the fork while @bear454 makes the custom view rendering a rails plugin that can live outside `blue-horizon` sources.

**Comment by stefanotorresi** [comment]
@arbulu89 ready for review.

**Comment by bear454** [comment]
Namespacing will be affected a bit by https://github.com/SUSE-Enceladus/blue-horizon/pull/176 .

File: app/helpers/sap_azure_plan_helper.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+# helper for rendering `terraform show` output as a plan
+module SapAzurePlanHelper
```
Comment (arbulu89) [inline app/helpers/sap_azure_plan_helper.rb]: I guess with this new PR: https://github.com/SUSE-Enceladus/blue-horizon/pull/176

This code could go in `CustomHelper`, right?

File: vendor/customization.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
     "allow_custom_instance_type": false,
     "show_instance_type_tip": false,
-    "cloud_framework": "Azure"
+    "cloud_framework": "Azure",
+    "override_views": "true"
```
Comment (arbulu89) [inline vendor/customization.json:5]: Is this variable only for spec testing?

File: vendor/views/plans/_plan.haml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      %dd.col-8= resources.dig 'module.bluehorizon.azurerm_virtual_network.mynet[0]', 'values', 'name'
+      %dt.col-4 Address range:
+      %dd.col-8= resources.dig('module.bluehorizon.azurerm_virtual_network.mynet[0]', 'values', 'address_space').join(',')
+    %dl.row.network
+      %dt.col-4 Name:
+      %dd.col-8= resources.dig 'module.bluehorizon.azurerm_subnet.mysubnet[0]', 'values', 'name'
+      %dt.col-4 Address range:
+      %dd.col-8= resources.dig('module.bluehorizon.azurerm_subnet.mysubnet[0]', 'values', 'address_prefixes').join(',')
+
+  %div.col.bastion
```
Comment (arbulu89) [inline vendor/views/plans/_plan.haml:90]: The iscsi server would be missing too.

**Review by arbulu89 — COMMENTED** [review]
Hi @stefanotorresi ,
Could we add the iscsi server information too?

Besides that, I think that the network information could come before the machines description.
In any case, as first approximation (and before getting some feedback from UX engineers) I like the view.

File: vendor/customization.json (hunk 2 above)
Comment (stefanotorresi) [reply vendor/customization.json:5]: nope, that enables the view globally.

File: vendor/views/plans/_plan.haml (hunk 3 above)
Comment (stefanotorresi) [reply vendor/views/plans/_plan.haml:90]: right! I will add it right away.

File: app/helpers/sap_azure_plan_helper.rb (hunk 1 above)
Comment (stefanotorresi) [reply app/helpers/sap_azure_plan_helper.rb]: exactly! I just merged that so i'm gonna sync the fork and rebase.

**Comment by stefanotorresi** [comment]
### updated screenshots

HA enabled:
Screenshot from 2020-11-13 10-53-37

HA disabled:
Screenshot from 2020-11-13 11-02-58

**Review by arbulu89 — APPROVED** [review]
@stefanotorresi Great! Fastastic job!
We can go with this as 1st version

**Pull request merged**